pkg/proxy/proxy.go

```diff
@@ -146,12 +147,25 @@ func (p *Proxy) allocatePort() (uint16, error) {
 
 var gcOnce sync.Once
 
+// RevertFunc is a function returned by a successful function call, which
+// reverts the side-effects of the initial function call. A call that returns
+// an error should return a nil RevertFunc.
+type RevertFunc func() error
```

Comment:
if RevertFunc and FinalizeFunc were in their own packages, you wouldn't have to duplicate the types in proxy / endpoint packages.

Reply:
Good idea, it's much cleaner. Done.